### PR TITLE
Write a readme on an example build workflow in gulp

### PR DESCRIPTION
**Pull request by hughesd22**
# References Issue #9

### CODE REVIEW

File: topics/gulp-build-workflow.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Creating a Build Workflow in Gulp
+In this guide, you will learn a bit about building a gulp workflow for optimizing and deploying web apps.
+
+Table of Contents
```
Comment (markmichon) [inline topics/gulp-build-workflow.md]: The table of contents items can be linked to their respective sections by creating in-page links. The markdown docs may have details, but headings on github will turn into linkable points via `#headingname`.

File: topics/gulp-build-workflow.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+Having a dist/build folder is imperative for deploying web apps efficiently. It allows you to compile all of your complete and usable code into a folder structure ready to be hosted for the world to see. It's a pretty simple concept, but it makes your work as a developer so much easier.
+
+Here is a an example folder structure of a web app being developed:
+
+```
+|- app/
+    |- css/
+    |- fonts/
+    |- images/
+    |- index.html/
```
Comment (markmichon) [inline topics/gulp-build-workflow.md]: Make sure not to end file names with a `/`, it implies they are directories.

File: topics/gulp-build-workflow.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+Just like before, we must run the task in order for the css files to be minified:
+```
+$ gulp build-css
+```
+
+Useful, eh?
+
+## Image Optimization
```
Comment (markmichon) [inline topics/gulp-build-workflow.md]: This section looks leftover. Is it intentionally empty?

File: topics/gulp-build-workflow.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+## Other Useful Plugins
+
+Now that you have a basic build workflow, you can branch out into more complicated (and customized) build workflows using whichever gulp plugins are applicable to your project. Some useful ones to keep in mind are:
+* gulp-concat (Concatenates JavaScript files)
+* gulp-minify-css (Another plugin to minify CSS files)
+* gulp-useref (Concatenates any number of CSS and JavaScript files into a single file)
+
+## Additional Resources and Reading
```
Comment (markmichon) [inline topics/gulp-build-workflow.md]: It would be great if the resource links were in an unordered list.

**Comment by hughesd22** [comment]
I made the requested changes and polished the readme

**Comment by hughesd22** [comment]
I made the requested changes and polished the readme. Let me know if you spot any typos that I may be blind to.

**Pull request merged**